src/Microsoft.OData.Core/PropertyAndAnnotationCollector.cs

```diff
@@ -264,7 +262,7 @@ internal void AddODataScopeAnnotation(string annotationName, object annotationVa
         internal void AddCustomScopeAnnotation(string annotationName, object annotationValue)
         {
             Debug.Assert(!string.IsNullOrEmpty(annotationName));
-            Debug.Assert(!JsonLight.ODataJsonLightReaderUtils.IsODataAnnotationName(annotationName));
+            Debug.Assert(!Json.ODataJsonReaderUtils.IsODataAnnotationName(annotationName));

```

Comment:
```suggestion
            Debug.Assert(!ODataJsonReaderUtils.IsODataAnnotationName(annotationName));
```
You already have a `Microsoft.OData.Json` using.